wordpress-mobile/WordPress-iOS
# [This Week Widget] Add functionality

Ref #13044

This adds full functionality to the This Week widget.

There are a few outstanding UI issues, which will be addressed in a follow-up PR:
- The footer jumps when expanding the widget.
- When the expanded widget is refreshed (ex: scrolling away from then back to Today):
  - The first two rows show the empty view while being refreshed.
  - The footer jumps.

To test:

---
- Start with a fresh install, or be logged out of the app.
- Add the `WordPress This Week` widget to your Today view.
  - Verify the unconfigured view is displayed.
  - Verify the widget cannot be expanded.
  - Verify tapping the widget launches the app.

<img width="400" alt="unconfigured" src="https://user-images.githubusercontent.com/1816888/71210672-a9de0900-226a-11ea-85b7-09571f5d4aba.png">

---
- Log into the app.
- Go to Stats > Widgets > `Use this site`.
  - Verify the This Week widget displays data for the selected site, 2 days in compact and 7 in expanded.
  - Verify the stats match the Days' chart data for each day.
  - Verify the widget can be expanded/collapsed.
  - Verify the footer is _not_ shown in compact, _is_ shown in expanded.

<img width="400" alt="app_stats" src="https://user-images.githubusercontent.com/1816888/71210724-c417e700-226a-11ea-83be-92c9866c3753.png">

<img width="400" alt="compact" src="https://user-images.githubusercontent.com/1816888/71210729-ca0dc800-226a-11ea-9be3-1bd80ac5db23.png">

<img width="400" alt="expanded" src="https://user-images.githubusercontent.com/1816888/71210732-cda14f00-226a-11ea-8a56-511ba3560f8a.png">

---
- Return to the app, and select a different site.
- Go to Stats > Widgets > `Use this site`.
- Verify the This Week widget updates accordingly.

---
PR submission checklist:
